### PR TITLE
Fix Clazy checks and make them blocking for CI

### PR DESCRIPTION
**Pull request by troopa81**
This PR fixes Clazy check (the ones we target in *CLAZY_BASE_CHECKS* cmake variable) and make them blocking for the CI.

I deleted the following checks: 
- *lambda-in-connect*: I think we will get only false positives because we use it with with a QEventLoop::exec() so the captured variable never reach end of scope before the lambda is called
- *qcolor-from-literal*: It crashes clang with clazy 1.11, so I commented for now, but maybe in the future we could add it again
- *qobject-macro*: I think it's useless to add a QOBJECT macro all the time because it increase compile time and it's not needed for all classes
- *non-pod-global-static*. It's temporary, I plan to revive it in a separate PR.

Regarding *skipped-base-class*, I added a skip instruction in code ([here](lien?) and all classes extending QgsBasicNumericFormat) because it would change behavior (sortKey is 100 in QgsNumericFormat vs 1 in QgsBasicNumericFormat). I'm not sure this is intended behavior? @nyalldawson ?

### CODE REVIEW

File: tests/src/analysis/testqgsprocessingalgspt1.cpp [hunk 1]
```diff
@@ -513,7 +513,7 @@ void TestQgsProcessingAlgsPt1::rasterLayerProperties()
   QCOMPARE( results.value( QStringLiteral( "EXTENT" ) ).toString(), QStringLiteral( "0.0000000000000000,0.0000000000000000 : 4.0000000000000000,4.0000000000000000" ) );
   QCOMPARE( results.value( QStringLiteral( "PIXEL_WIDTH" ) ).toDouble(), 1.0 );
   QCOMPARE( results.value( QStringLiteral( "PIXEL_HEIGHT" ) ).toDouble(), 1.0 );
-  QCOMPARE( results.value( QStringLiteral( "CRS_AUTHID" ) ).toString(), QStringLiteral( "" ) );
+  QCOMPARE( results.value( QStringLiteral( "CRS_AUTHID" ) ).toString(), QStringLiteral( "" ) ); // clazy:exclude=empty-qstringliteral
```
Comment (elpaso) [inline tests/src/analysis/testqgsprocessingalgspt1.cpp:516]: I think you can actually replace QStringLiteral( "" ) with QString().

**Comment by elpaso** [comment]
Nice!

**Comment by nyalldawson** [comment]
@troopa81 

Great stuff, I've long wanted this on our CI :+1:  :partying_face: 

> qobject-macro: I think it's useless to add a QOBJECT macro all the time because it increase compile time and it's not needed for all classes

Actually it IS needed for ALL QObject subclasses -- Qt documentation is quite clear about this:

" We strongly recommend the use of this macro in all subclasses of QObject regardless of whether or not they actually use signals, slots and properties, since failure to do so may lead certain functions to exhibit strange behavior. " (from https://doc.qt.io/qt-5/qobject.html ). 

I'd suggest resurrecting this one accordingly.

> Regarding skipped-base-class, I added a skip instruction in code ([here](https://github.com/qgis/QGIS/pull/lien?) and all classes extending QgsBasicNumericFormat) because it would change behavior (sortKey is 100 in QgsNumericFormat vs 1 in QgsBasicNumericFormat). I'm not sure this is intended behavior? @nyalldawson ?

Yes, it's intended. We want basic numeric format to always appear first, but its various subclasses don't also get preference and instead should just be sorted using the default value (so that they get alphabetical sorting). We could hardcode the "100" value from QgsNumericFormat::sortKey() in the QgsBasicNumericFormat subclasses sortKey overrides, but to me it's a bit cleaner leaving this constant value only defined in one place. A good alternative would be to add a `constexpr int DEFAULT_SORT_KEY` protected member in QgsNumericFormat and use that accordingly.

File: tests/src/analysis/testqgsprocessingalgspt1.cpp (hunk 1 above)
Comment (troopa81) [reply tests/src/analysis/testqgsprocessingalgspt1.cpp]: Yes. Fixed

**Comment by troopa81** [comment]
> I'd suggest resurrecting this one accordingly.

OK, I propose to treat them in a separate PR, I think this one is already big enough

> A good alternative would be to add a constexpr int DEFAULT_SORT_KEY protected member in QgsNumericFormat and use that accordingly.

Fixed [aa3be46](https://github.com/qgis/QGIS/pull/48457/commits/aa3be468d7c8eb24e6cd5a757dd2cef5d13db6f7)

**Review by nyalldawson — APPROVED** [review]

**Comment by troopa81** [comment]
Since I added a parameter *with-clazy* to build job, I think that the expected job name need to be [updated](https://github.community/t/expected-waiting-for-status-to-be-reported/18001) in project settings. 

Can someone update the settings, so this PR could be merged?

**Pull request merged**